### PR TITLE
Feature/test with statechange

### PR DESCRIPTION
**Pull request by tuansondinh**
tests regarding connect, subscribe and executing commands

Service/procedure parameters, current procedure will be tackled in another branch.
BEFORE (JUNE)
image
AFTER(AUGUST)
tc

### CODE REVIEW

**Comment by tuansondinh** [comment]
@julorenz could you please review again? some TODOs will be tackled in another branch.

File: src/modularPlantManager/ModularPlantManager.spec.ts [hunk 1]
```diff
@@ -258,52 +237,14 @@ describe('ModularPlantManager', () => {
...
-			const service = modularPlantManager.getService('Dose', 'Fill');
-			expect(service).to.be.instanceOf(Service);
-			expect(service.name).to.equal('Fill');
-			expect(() => modularPlantManager.getService('Dose', 'NoService')).to.throw();
-			expect(() => modularPlantManager.getService('NoPEA', 'NoService')).to.throw();
-
-			await expect(modularPlantManager.removePEAController('something')).to.be.rejectedWith('PEAController with id something not found');
-		});*/
-
-		it('should prevent removing a protected PEAController', async () => {
```
Comment (julorenz) [inline src/modularPlantManager/ModularPlantManager.spec.ts:270]: Why did you remove this part?

File: src/modularPlantManager/pea/PEAController.ts [hunk 2]
```diff
@@ -309,7 +309,6 @@ export class PEAController extends (EventEmitter as new() => PEAEmitter) {
 			if (service.isCommandExecutable(ServiceCommand.pause)) {
 				return await service.executeCommandAndWaitForStateChange(ServiceCommand.pause);
 			} else {
-				//TODO is this legit? should we throw an error here?
```
Comment (julorenz) [inline src/modularPlantManager/pea/PEAController.ts]: Whats the answer to that question?

Comment (tuansondinh) [reply src/modularPlantManager/pea/PEAController.ts]: yes, we should throw error !

File: src/modularPlantManager/ModularPlantManager.spec.ts (hunk 1 above)
Comment (tuansondinh) [reply src/modularPlantManager/ModularPlantManager.spec.ts:270]: Hey, the achema json file doesn't exist anymore, so this test doesn't work. Should I keep it as a TODO?

Comment (tuansondinh) [reply src/modularPlantManager/ModularPlantManager.spec.ts:270]: I have added it again

**Review by julorenz — APPROVED** [review]
